lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart

```diff
@@ -0,0 +1,333 @@
+                            : unknown == true
+                                ? DomainType.unknown
+                                : DomainType.valid,
+                  );
+                },
+                fullscreenDialog: true));
+
+        if (userhasConfirmed != null && userhasConfirmed) {
+          //Accepted
+          try {
+            if (mounted) {
+              Navigator.of(context).pop();
+              final l10n = l10nOf(context);
+              _globalSnackBar.show(l10n.wcConnectionApproved);
+            }
+          } catch (e) {
+            _globalSnackBar.showError(e.toString());
+            setState(() {
+              isLoading = false;
+              if (pairingTimer != null) pairingTimer!.cancel();
+            });
+          }
+        } else if (userhasConfirmed == null) {
+          //Rejected
+          setState(() {
+            isLoading = false;
+            if (pairingTimer != null) pairingTimer!.cancel();
+          });
+          await walletConnectService.web3Wallet?.rejectSession(
+              id: wcPairingId!,
+              reason: Errors.getSdkError(Errors.USER_REJECTED).toSignError());
+
+          if (mounted) {
+            Navigator.of(context).pop();
+            final l10n = l10nOf(context);
+            _globalSnackBar.showError(l10n.wcConnectionRejected);
+          }
+        } else {
+          // false : expired
+          setState(() {
+            isLoading = false;
+            if (pairingTimer != null) pairingTimer!.cancel();
+          });
+
+          if (mounted) {
+            final l10n = l10nOf(context);
+            Navigator.of(context).pop();
+            _globalSnackBar.show(l10n.wcConnectionProposalTimeout);
+          }
+        }
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    if (sessionProposalSubscription != null) {
+      sessionProposalSubscription!.cancel();
+    }
+    if (sessionProposalErrorSubscription != null) {
+      sessionProposalErrorSubscription!.cancel();
+    }
+    pairingTimer?.cancel();
+    urlController.dispose();
+    super.dispose();
+  }
+
+  pasteAndProceed() async {
+    final clipboardData = await Clipboard.getData(Clipboard.kTextPlain);
+    String? clipboardText = clipboardData?.text;
+    proceedHandler(clipboardText);
+  }
+
+  void proceedHandler(String? url) async {
+    final l10n = l10nOf(context);
+    if (validateWalletConnectURL(url) != null) {
+      _globalSnackBar.showError(validateWalletConnectURL(url)!);
+      return;
+    }
+
+    setState(() {
+      isLoading = true;
+    });
+
+    if (!walletConnectService.web3Wallet!.core.relayClient.isConnected) {
+      await walletConnectService.web3Wallet!.core.relayClient.connect();
+    }
+    try {
+      PairingInfo pairResult = await walletConnectService.pair(Uri.parse(url!));
+
+      if (walletConnectService
+          .sessionPairingTopicAlreadyExists(pairResult.topic)) {
+        pairingTimer = Timer(const Duration(seconds: 10), () {
```

Comment:
What is the logic of using those 10 seconds here? It loads too long if the URL is already used.

Reply:
Older version could issue a success to a pairing topic even after it reported failingly that it already exists. Not relevant any more in reown. removed it